storage/badger/receipts.go

```diff
@@ -142,8 +143,78 @@ func (r *ExecutionReceipts) Index(blockID, receiptID flow.Identifier) error {
 	})
 }
 
+func (r *ExecutionReceipts) IndexByBlockIDAndExecutionID(blockID, executionID, receiptID flow.Identifier) error {
```

Comment:
As mentioned on the call, I think it's worth consolidating this API a bit when these changes are introduced through a spork rather than a hotfix. The `executorID` being included in the key bytes is an implementation detail to support a one->many index -- ideally this implementation detail wouldn't be included in the API. When we don't need to worry about data migration post-spork, we can have a single `blockID->receiptID` index that supports multiple receipts for a single block and can be used by both exe and access nodes.